libs/external-router/driver/testing/src/testing.service.ts

```diff
@@ -0,0 +1,37 @@
+import { Injectable, Inject } from '@angular/core';
+import { Observable, of, throwError } from 'rxjs';
+
+import { DaffResolvableRoute } from '@daffodil/external-router';
+import { DaffExternalRouterDriverInterface } from '@daffodil/external-router/driver';
+import {
+	DaffExternalRouterDriverTestingConfig,
+	DAFF_EXTERNAL_ROUTER_DRIVER_TESTING_CONFIG,
+} from './config';
+
+/**
+ * The DaffExternalRouterTestingDriver is responsible for translating an
+ * arbitrary URI into a DaffResolvableRoute in testing environments.
+ */
+@Injectable({
+	providedIn: 'root',
+})
+export class DaffExternalRouterTestingDriver
+	implements DaffExternalRouterDriverInterface {
+	constructor(
+		@Inject(DAFF_EXTERNAL_ROUTER_DRIVER_TESTING_CONFIG)
+		private testingConfiguration: DaffExternalRouterDriverTestingConfig = {},
+	) {}
+
+	resolve(url: string): Observable<DaffResolvableRoute> {
+		if (!this.testingConfiguration[url]) {
+			return throwError(`\
+The route '${url}' wasn't provided with a matching type by the testing driver. \
+Did you configure the available route types in DaffExternalRouterDriverTestingModule.forRoot()`);
```

Comment:
I think this should read `with DaffExternalRouterDriverTestingModule.forRoot(config)` to reduce possible confusion

Reply:
https://github.com/graycoreio/daffodil/pull/1240/commits/e84f901398d30c4af371fb257714725339f3640a

Reply:
I meant for the important part of that to be the "config" that is passed in:
`with DaffExternalRouterDriverTestingModule.forRoot(C_O_N_F_I_G)`

Reply:
be4a79e3